### PR TITLE
Rename dependency is_resolved to is_pinned

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
Renaming the dependency attribute is_resolved to is_pinned to more accurately represent this attribute accurately. This is more relevant after the changes in nexB/scancode.io#1066

Reference: https://github.com/nexB/scancode.io/issues/1066

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [x] Updated documentation pages (if applicable)
* [x] Updated CHANGELOG.rst (if applicable)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -34,6 +34,20 @@ v33.0.0 (next next, roadmap)
...
+- Output Format Version updated to 4.2.0 (major version bump)
+- Dependency attribute rename: ``is_resolved`` renamed to ``is_pinned``
+  See https://github.com/nexB/scancode-toolkit/pull/3888 for more details.
+
+Changes in Output Data Structure:
+
+- The data structure of the JSON output has changed for
+  dependencies at file level package_data, and at top-level.
+  Note that the change is a modification to the JSON output,
+  so we have a major version bump ``3.2.0`` to ``4.2.0``:
```
Comment (pombredanne) [inline CHANGELOG.rst]:
```suggestion
  so we have a major version bump ``3.2.0`` to ``4.0.0``:
```

File: src/scancode_config.py [hunk 2]
```diff
@@ -141,7 +141,7 @@ def _create_dir(location):
 
 # See https://github.com/nexB/scancode-toolkit/issues/2653 for more information
 # on the data format version
-__output_format_version__ = '3.2.0'
+__output_format_version__ = '4.2.0'
```
Comment (pombredanne) [inline src/scancode_config.py]:
```suggestion
__output_format_version__ = '4.0.0'
```

File: etc/scripts/sch2js/doc/DependentPackage-json-schema.json [hunk 3]
```diff
@@ -63,12 +63,12 @@
         }
       ]
     }, 
-    "is_resolved": {
+    "is_pinned": {
```
Comment (pombredanne) [inline etc/scripts/sch2js/doc/DependentPackage-json-schema.json:66]: You may want to regen the schema or to drop these schema entirely in the future

**Review by pombredanne — APPROVED** [review]
Here are some nit for your consideration

File: CHANGELOG.rst (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply CHANGELOG.rst]: Added, thanks!

File: src/scancode_config.py (hunk 2 above)
Comment (AyanSinhaMahapatra) [reply src/scancode_config.py]: Added, thanks!

**Comment by AyanSinhaMahapatra** [comment]
Thanks @pombredanne, resolved your comments and merged!

**Pull request merged**